soroban-env-host/src/host.rs

```diff
@@ -2281,6 +2271,35 @@ impl VmCallerEnv for Host {
     fn dummy0(&self, vmcaller: &mut VmCaller<Self::VmUserState>) -> Result<RawVal, Self::Error> {
         Ok(().into())
     }
+
+    fn require_auth(
+        &self,
+        vmcaller: &mut VmCaller<Self::VmUserState>,
+        address: Object,
+        args: Object,
+    ) -> Result<RawVal, Self::Error> {
+        let addr = self.visit_obj(address, |addr: &ScAddress| Ok(addr.clone()))?;
+
+        Ok(self
+            .0
+            .authorization_manager
+            .borrow_mut()
+            .require_auth(
+                self,
+                address.get_handle(),
```

Comment:
Using the handle to track different `require_auth` calls for the same `ScAddress` in recording mode looks like a footgun to me. If I understood @graydon at standup correctly, we are considering making these objects unique (so handles will be the same). A change like that would break auth. This shouldn't hold up the release, but it's something we should discuss for the next auth update.

Reply:
> If I understood @graydon at standup correctly, we are considering making these objects unique (so handles will be the same).

I'm not sure I understood, would we somehow have a single set of objects and do the lookups? That doesn't sound efficient...

>  A change like that would break auth. 
FWIW this is used in the recording mode only. But if we somehow would try to deduplicate objects, then `Address` host object type would need an additional level of indirection in order to distinguish between different objects.